internal/delete/delete_item_action_handler.go

```diff
@@ -0,0 +1,218 @@
+	// just do the backup deletion without worrying about plugins.
+	if (ctx.resolvedActions == nil || len(ctx.resolvedActions) == 0) && err == nil {
+		ctx.Log.Debug("No delete item actions present, proceeding with rest of backup deletion process")
+		return nil
+	} else if err != nil {
+		return errors.Wrapf(err, "error resolving actions")
+	}
+
+	// get items out of backup tarball into a temp directory
+	dir, err := archive.NewExtractor(ctx.Log, ctx.Filesystem).UnzipAndExtractBackup(ctx.BackupReader)
+	if err != nil {
+		return errors.Wrapf(err, "error extracting backup")
+
+	}
+	defer ctx.Filesystem.RemoveAll(dir)
+	ctx.Log.Debugf("Downloaded and extracted the backup file to: %s", dir)
+
+	backupResources, err := archive.NewParser(ctx.Log, ctx.Filesystem).Parse(dir)
+	processdResources := sets.NewString()
+
+	ctx.Log.Debugf("Trying to reconcile resource names with Kube API server.")
+	// Transform resource names based on what's canonical in the API server.
+	for resource := range backupResources {
+		gvr, _, err := ctx.DiscoveryHelper.ResourceFor(schema.ParseGroupResource(resource).WithVersion(""))
+		if err != nil {
+			return errors.Wrapf(err, "failed to resolve resource into complete group/version/resource: %v", resource)
+		}
+
+		groupResource := gvr.GroupResource()
+
+		// We've already seen this group/resource, so don't process it again.
+		if processdResources.Has(groupResource.String()) {
+			continue
+		}
+
+		// Get a list of all items that exist for this resource
+		resourceList := backupResources[groupResource.String()]
+		if resourceList == nil {
+			// After canonicalization from the API server, the resources may not exist in the tarball
+			// Skip them if that's the case.
+			continue
+		}
+
+		// Iterate over all items, grouped by namespace.
+		for namespace, items := range resourceList.ItemsByNamespace {
+			nsLog := ctx.Log.WithField("namespace", namespace)
+			nsLog.Info("Starting to check for items in namespace")
+
+			// Filter applicable actions based on namespace only once per namespace.
+			actions := ctx.getApplicableActions(groupResource, namespace)
+
+			// Process individual items from the backup
+			for _, item := range items {
+				itemPath := archive.GetItemFilePath(dir, resource, namespace, item)
+
+				// obj is the Unstructured item from the backup
+				obj, err := ctx.unmarshal(itemPath)
+				if err != nil {
+					return errors.Wrapf(err, "Could not unmarshal item: %v", item)
+				}
+
+				itemLog := nsLog.WithField("item", obj.GetName())
+				itemLog.Infof("invoking DeleteItemAction plugins")
+
+				for _, action := range actions {
+					if !action.selector.Matches(labels.Set(obj.GetLabels())) {
+						continue
+					}
+					err = action.Execute(&velero.DeleteItemActionExecuteInput{
+						Item:   obj,
+						Backup: ctx.Backup,
+					})
+					// Since we want to keep looping even on errors, log them instead of just returning.
+					if err != nil {
+						itemLog.WithError(err).Error("plugin error")
+
```

Comment:
Yeah, it would. Let me do some testing/checking, because I _think_ the error returned will actually have the plugin's information passed back through gRPC, but I don't remember. I don't know if the name's directly available here.